// ufc-api/dist/v1/api.js
'use strict';

var sendRequest = require('../utils').sendRequest;

var ENDPOINT = 'http://ufc-data-api.ufc.com/api/v1/us/';

function uriFor(path) {
  return ENDPOINT + path;
}

var api = {
  health: function health(done) {
    sendRequest('GET', uriFor('health'), done);
  },
  news: function news(done) {
    sendRequest('GET', uriFor('news'), done);
  },
  article: function article(id, done) {
    sendRequest('GET', uriFor('news') + '/' + id, done);
  },
  media: function media(done) {
    sendRequest('GET', uriFor('media'), done);
  },
  mediaItem: function mediaItem(id, done) {
    sendRequest('GET', uriFor('media') + '/' + id, done);
  },
  titleHolders: function titleHolders(done) {
    sendRequest('GET', uriFor('fighters/title_holders'), done);
  },
  fighters: function fighters(done) {
    sendRequest('GET', uriFor('fighters'), done);
  },
  fighter: function fighter(id, done) {
    sendRequest('GET', uriFor('fighters') + '/' + id, done);
  },
  events: function events(done) {
    sendRequest('GET', uriFor('events'), done);
  },
  event: function event(id, done) {
    sendRequest('GET', uriFor('events') + '/' + id, done);
  },
  alerts: function alerts(done) {
    sendRequest('GET', uriFor('alerts'), done);
  }
};

module.exports = api;
