// ufc-api/dist/v3/api.js
'use strict';

var sendRequest = require('../utils').sendRequest;

var ENDPOINT = 'http://ufc-data-api.ufc.com/api/v3/us/';

function uriFor(path) {
  return ENDPOINT + path;
}

var api = {
  health: function health(done) {
    sendRequest('GET', uriFor('health'), done);
  },
  translations: function translations(id, done) {
    sendRequest('GET', uriFor('translations') + '/' + id, done);
  },
  streamingVideo: function streamingVideo(done) {
    sendRequest('GET', uriFor('live'), done);
  },
  liveContent: function liveContent(done) {
    sendRequest('GET', uriFor('live_content'), done);
  },
  chat: function chat(done) {
    sendRequest('GET', uriFor('chat'), done);
  },
  instructions: function instructions(done) {
    sendRequest('GET', uriFor('instructions'), done);
  },
  news: function news(done) {
    sendRequest('GET', uriFor('news'), done);
  },
  article: function article(id, done) {
    sendRequest('GET', uriFor('news') + '/' + id, done);
  },
  media: function media(done) {
    sendRequest('GET', uriFor('media'), done);
  },
  mediaItem: function mediaItem(id, done) {
    sendRequest('GET', uriFor('media') + '/' + id, done);
  },
  titleHolders: function titleHolders(done) {
    sendRequest('GET', uriFor('fighters/title_holders'), done);
  },
  fighters: function fighters(done) {
    sendRequest('GET', uriFor('fighters'), done);
  },
  fighter: function fighter(id, done) {
    sendRequest('GET', uriFor('fighters') + '/' + id, done);
  },
  fighterStatsFilterKeyValues: function fighterStatsFilterKeyValues(done) {
    sendRequest('GET', uriFor('fighters/stats_filter_key_values'), done);
  },
  fighterStatsByWeightClass: function fighterStatsByWeightClass(id, done) {
    sendRequest('GET', uriFor('fighters/stats/weight_class') + '/' + id, done);
  },
  fighterStatsByBirthCountry: function fighterStatsByBirthCountry(id, done) {
    sendRequest('GET', uriFor('fighters/stats/born_country') + '/' + id, done);
  },
  fighterNews: function fighterNews(id, done) {
    sendRequest('GET', uriFor('fighters') + '/' + id + '/news', done);
  },
  fighterMedia: function fighterMedia(id, done) {
    sendRequest('GET', uriFor('fighters') + '/' + id + '/media', done);
  },
  octagonGirls: function octagonGirls(done) {
    sendRequest('GET', uriFor('octagon_girls'), done);
  },
  octagonGirl: function octagonGirl(id, done) {
    sendRequest('GET', uriFor('octagon_girls') + '/' + id, done);
  },
  events: function events(done) {
    sendRequest('GET', uriFor('events'), done);
  },
  event: function event(id, done) {
    sendRequest('GET', uriFor('events') + '/' + id, done);
  },
  eventFights: function eventFights(id, done) {
    sendRequest('GET', uriFor('events') + '/' + id + '/fights', done);
  },
  eventMedia: function eventMedia(id, done) {
    sendRequest('GET', uriFor('events') + '/' + id + '/media', done);
  },
  eventHighlights: function eventHighlights(id, done) {
    sendRequest('GET', uriFor('events') + '/' + id + '/highlights', done);
  },
  fightVotes: function fightVotes(id, done) {
    sendRequest('GET', uriFor('fights') + '/' + id + '/votes', done);
  },
  fightGetVote: function fightGetVote(fightId, roundNum, fighterStatId, clientId, done) {
    sendRequest('GET', uriFor('fights') + '/' + fightId + '/round/' + roundNum + '/fighter/' + fighterStatId + '/vote/' + clientId, done);
  },
  fightPostVote: function fightPostVote(id, done) {
    sendRequest('POST', uriFor('fights') + '/' + id + '/vote', done);
  },
  roundScoring: function roundScoring(id, done) {
    sendRequest('GET', uriFor('events/round_scoring') + '/' + id, done);
  },
  discussEvent: function discussEvent(id, done) {
    sendRequest('GET', uriFor('events') + '/' + id + '/discuss', done);
  },
  eventTickets: function eventTickets(id, done) {
    sendRequest('GET', uriFor('events') + '/' + id + '/tickets', done);
  },
  watchEvent: function watchEvent(id, done) {
    sendRequest('GET', uriFor('events') + '/' + id + '/watch', done);
  },
  alerts: function alerts(done) {
    sendRequest('GET', uriFor('alerts'), done);
  }
};

module.exports = api;
